tests/countries/test_trinidad_and_tobago.py

```diff
@@ -0,0 +1,358 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see CONTRIBUTORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from unittest import TestCase
+
+from holidays.constants import OPTIONAL
+from holidays.countries.trinidad_and_tobago import TrinidadAndTobago, TT, TTO
+from tests.common import CommonCountryTests
+
+
+class TestTrinidadAndTobago(CommonCountryTests, TestCase):
+    @classmethod
+    def setUpClass(cls):
+        years = range(1963, 2050)
+        super().setUpClass(TrinidadAndTobago, years=years, years_non_observed=years)
+        cls.no_estimated_holidays = TrinidadAndTobago(years=years, islamic_show_estimated=False)
+        cls.optional_holidays = TrinidadAndTobago(categories=OPTIONAL, years=years)
+
+    def test_country_aliases(self):
+        self.assertAliases(TrinidadAndTobago, TT, TTO)
+
+    def test_no_holidays(self):
+        self.assertNoHolidays(TrinidadAndTobago(years=1962))
+        self.assertNoHolidays(TrinidadAndTobago(categories=OPTIONAL, years=1962))
+
+    def test_new_years_day(self):
+        name = "New Year's Day"
+        self.assertHolidayName(name, (f"{year}-01-01" for year in range(1963, 2050)))
+        dt = (
+            "2006-01-02",
+            "2012-01-02",
+            "2017-01-02",
+            "2023-01-02",
+        )
+        self.assertHolidayName(f"{name} (observed)", dt)
+        self.assertNoNonObservedHoliday(dt)
+
+    def test_good_friday(self):
+        name = "Good Friday"
+        self.assertHolidayName(
+            name,
+            "2020-04-10",
+            "2021-04-02",
+            "2022-04-15",
+            "2023-04-07",
+            "2024-03-29",
+            "2025-04-18",
+        )
+        dt = ("2018-04-03",)
+        self.assertHolidayName(f"{name} (observed)", dt)
+        self.assertHolidayName(name, range(1963, 2050))
```

Comment:
```suggestion
        self.assertHolidayName(name, range(1963, 2050))
        dt = (
            "2018-04-03",
            "2029-04-03",
        )
        self.assertHolidayName(f"{name} (observed)", dt)
        self.assertNoNonObservedHoliday(dt)
```
`self.assertHolidayName(name, ...` is a part of general holiday checks, it should be after them, not after observed holidays checks. (Update this for rest of holidays)